Makefile

```diff
@@ -1351,6 +1351,7 @@ distclean: clean
 	rm -f boot/ocamlrun boot/ocamlrun$(EXE) boot/camlheader \
 	      boot/ocamlyacc boot/*.cm* boot/libcamlrun.$(A)
 	rm -f config/Makefile byterun/caml/m.h byterun/caml/s.h
+	rm -f config/auto-aux/hashbang4
```

Comment:
I think you could remove the file in `configure` directly, right after HASHBANGSCRIPTS is set.